src/simpleclient.ts

```diff
@@ -0,0 +1,561 @@
+    );
+  }
+
+  // todo: is the name misleading? should it be just
+  // getOrdersAccountForSymbol or getAccountForSymbol?
+  private async getOpenOrdersAccountForSymbol(
+    marketSymbol: SpotMarketSymbol,
+  ): Promise<OpenOrders[]> {
+    if (Object.keys(this.spotMarketMappings).indexOf(marketSymbol) === -1) {
+      throw new Error(`unknown spot market ${marketSymbol}`);
+    }
+    const marketAddress = this.spotMarketMappings[marketSymbol];
+    const market = this.markets.find((market) =>
+      market.publicKey.equals(marketAddress),
+    );
+
+    const marketIndex = this.mangoGroup.getMarketIndex(market!);
+    const marginAccounts = await this.getMarginAccountsForOwner();
+    return marginAccounts
+      .map((marginAccount) => marginAccount.openOrdersAccounts[marketIndex])
+      .filter(notEmpty);
+  }
+
+  private async cancelOrder(
+    marginAccount: MarginAccount,
+    market: Market,
+    order: Order,
+  ): Promise<TransactionSignature> {
+    return this.client.cancelOrder(
+      this.connection,
+      this.programId,
+      this.mangoGroup,
+      marginAccount,
+      this.payer,
+      market,
+      order,
+    );
+  }
+
+  private async cancelOrdersForMarginAccount(
+    marginAccount: MarginAccount,
+    symbol?: SpotMarketSymbol,
+    orderId?: string,
+  ) {
+    let orders;
+    let market;
+
+    if (symbol === undefined) {
+      for (const spotMarketSymbol of Object.keys(this.spotMarketMappings)) {
+        market = this.getMarketForSymbol(spotMarketSymbol);
+        orders = await this.getOpenOrders(spotMarketSymbol);
+        await orders.map((order) =>
+          this.cancelOrder(marginAccount, market, order),
+        );
+      }
+      return;
+    }
+
+    market = this.getMarketForSymbol(symbol!);
+    orders = await this.getOpenOrders(symbol!);
+    // note: orderId could not even belong to his margin account
+    // in that case ordersToCancel would be empty
+    const ordersToCancel = orderId
+      ? orders.filter((o) => o.orderId.toString() === orderId)
+      : orders;
+
+    await Promise.all(
+      ordersToCancel.map((order) =>
+        this.cancelOrder(marginAccount, market, order),
+      ),
+    );
+  }
+
+  /// public
+
+  async placeOrder(
+    symbol: SpotMarketSymbol,
+    side: 'buy' | 'sell',
+    quantity: number,
+    price: number,
+  ): Promise<TransactionSignature> {
+    if (!symbol.trim()) {
+      throw new Error(`invalid symbol ${symbol}`);
+    }
+    if (!Number.isFinite(quantity) || quantity <= 0) {
+      throw new Error(`invalid quantity ${quantity}`);
+    }
+    if (!Number.isFinite(price) || price <= 0) {
+      throw new Error(`invalid price ${price}`);
+    }
+
+    const market = this.getMarketForSymbol(symbol);
+
+    const marginAccounts = await this.getMarginAccountsForOwner();
+    // todo: rather choose margin account with largest balance for base or
+    // quote currency based on trade side
+    const marginAccountWithLargestBalance = marginAccounts[0];
```

Comment:
most users will have exactly one margin account (which this client should create on demand imo). users with multiple accounts will have those for the purpose of managing risk or for participation in a trading competition. the users with multiple accounts most likely want to have the control "where their order is placed". what is the scenario you were thinking of solving with this logic?

what do you think about a scoped / more stateful design for the client, which can only be used to work with a single margin account at a time?

Reply:
I agree, I changed the code to grab by default largest margin account the user has and implicitly use that. Other than this I create a margin account inline implicitly if none exists